plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/blobstore/AzureBlobContainer.java

```diff
@@ -72,14 +72,14 @@ public InputStream readBlob(String blobName) throws IOException {
         logger.trace("readBlob({})", blobName);
 
         if (!blobExists(blobName)) {
```

Comment:
@imotov: Azure is a little too lenient about this unfortunately (at least in testing), hence necessitating this check.  FWIW, @abeyad could remove it and run the unit tests again to see whether we still get those failures.